core/src/main/java/com/linecorp/armeria/internal/common/util/MinifiedBouncyCastleProvider.java

```diff
@@ -139,15 +130,28 @@ public boolean hasAlgorithm(String type, String name) {
     @Override
     public void addAlgorithm(String key, String value) {
         checkState(!containsKey(key), "duplicate algorithm: %s", key);
+        if (value.contains(".bouncycastle.")) {
+            // Very likely to be a class name.
+            assertClassName(value);
+        }
         put(key, value);
     }
 
     @Override
     public void addAlgorithm(String type, ASN1ObjectIdentifier oid, String className) {
+        assertClassName(className);
         addAlgorithm(type + '.' + oid, className);
         addAlgorithm(type + ".OID." + oid, className);
     }
 
+    private static void assertClassName(String className) {
+        // Bouncy Castle sometimes uses a hard-coded class name when configuring itself.
+        // Such hard-coded names may or may not be updated during the class relocation process
+        // depending on how and what relocation tool we use, so this `assert` will help us
+        // notice when the relocation did not work as expected.
+        assert className.startsWith(BC_PACKAGE_PREFIX) : "Unexpected BouncyCastle class name: " + className;
```

Comment:
Then, `IllegalArgumentException`?
We normally use assert statements when we believe it's true.